core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -118,14 +119,16 @@ private[spark] class MesosSchedulerBackend(
     val command = CommandInfo.newBuilder()
       .setEnvironment(environment)
     val uri = sc.conf.get("spark.executor.uri", null)
+    val executorBackendName = classOf[MesosExecutorBackend].getName
     if (uri == null) {
-      val executorPath = new File(executorSparkHome, "/sbin/spark-executor").getCanonicalPath
+      val executorPath = new File(executorSparkHome, s"/bin/spark-class $executorBackendName")
+        .getCanonicalPath
```

Comment:
This is actually weird, why are we including `$executorBackendName` in the filename? We should just look for spark-class. I'm not even sure getCanonicalPath will work on a file that doesn't exist